### ISSUE TITLE
Certain magnets forcing Real-Debrid to re-download

### ISSUE THREAD

**Issue by logtho** [rogerfar/rdt-client#31]
I have at least two magnet links that seem to be saved on Real-Debrid just fine - but when putting them into RDC, it tells Real-Debrid to start downloading the entire torrent again.

I have the "Only download available files on RealDebrid" option enabled, and I have tried the same magnet links with "Minimum file size to download" set to 10 MBs, 0 MBs, and 100 MBs all with the same outcome.

I'm happy to share these magnet links if wanted.

**Comment by logtho** [comment]
Update:
I got one of these magnets downloaded through RDC, and it only downloaded 1 file out of 364, despite most of (if not all) 364 being available to download when I put the magnet link directly onto Real-Debrid.
The other has 183 files however only 1/3 of them are needed (each episode has a .mkv, .nfo, and .jpg, but only the .mkv is needed), and downloading to Real-Debrid incredibly slowly.

**Comment by rogerfar** [comment]
Strange, can you send me those links in a pastebin?

**Comment by rogerfar** [comment]
So I tried it with the large torrent of 364 files and that is correct, all of those 364 files are MKV files and are available on RD. That last torrent is all sorts of broken I think, I'm getting odd errors from RD when trying to unrestrict and it only has 3 files available.

Now I added a feature in RDT client where you can view the available files, it's in the latest version under "Add Torrent", there's a button to view all the files available on RD.

Another thing I changed is the unrestrict part, when you have large torrents it will unrestrict all the links as soon as the torrent is done on RD, but it might be a while before you start downloading each file, so it can timeout.

Now it will unrestrict the download link only when it's about to download.

**Comment by logtho** [comment]
Thanks for the update - that did indeed fix one of those magnet links (and I really like having the ability to see available files). I'm still having issues with the other two though; when I put them in RD's site, it shows that all are available. But when I put them in RDT client, it starts downloading the torrent to RD instead of just downloading them from RD to my computer like normal.

You say that the third link is broken, I assume there's something weird going on with the RD API or something? What about the first link? Is that one giving the same problem?

**Comment by rogerfar** [comment]
When I check that first torrent it shows me a list of mkv's and nfo's. So that means they should be available in RD.

**Comment by rogerfar** [comment]
I tried it with a newer torrent and I think I see what you mean. I think what happens is that Real Debrid reports certain files as available, even though they are still being downloaded. It basically reports that the torrent has already been added.

Unfortunately the RD API doesn't return the status of an available file, just the torrent hash. For example, for your first torrent I returns this (raw): https://pastebin.com/Ldw2C4h8
